televisionbingecalculator/src/main/java/com/vanillax/televisionbingecalculator/app/kotlin/viewmodels/LandingActivityViewModel.kt

```diff
@@ -1,41 +1,42 @@
 package com.vanillax.televisionbingecalculator.app.kotlin.viewmodels
 
+import android.util.Log
 import androidx.databinding.ObservableField
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
 import com.vanillax.televisionbingecalculator.app.kotlin.enum.SearchType
 import com.vanillax.televisionbingecalculator.app.kotlin.network.TVBCLoggerService
 import com.vanillax.televisionbingecalculator.app.kotlin.network.TheMovieDBService
 import com.vanillax.televisionbingecalculator.app.kotlin.network.response.QueryResponse
 import com.vanillax.televisionbingecalculator.app.serverapi.TVBCLogger.SearchTerm
 import io.reactivex.disposables.Disposable
+import io.reactivex.functions.Consumer
 
 
 /**
  * Created by mitchross on 4/14/18.
  */
+//TODO add tests around livedata and this viewmodel
+//TODO make this viewmodel lifecycleobserver to handle n/w calls without activity telling it
 class LandingActivityViewModel(
-        theMovieDBService: TheMovieDBService,
-        tvbcLoggerService: TVBCLoggerService
-) {
-
-    interface LandingActivityViewModelInterface {
-        fun updateShowList(queryResponse: QueryResponse, searchType: SearchType)
-        fun onTouch(id: Int, url: String, title: String)
-        fun error(error: String?)
-    }
+    theMovieDBService: TheMovieDBService,
+    tvbcLoggerService: TVBCLoggerService
```

Comment:
Not an issue here but we should switch to a repository pattern to abstract these services away from the ViewModel

Reply:
yes agree already have this as a todo.